src/hydro_forces.cpp

```diff
@@ -353,58 +353,49 @@ double TestHydro::setVelHistory(double val, int step, int b_num, int index) {
 *******************************************************************************/
 std::vector<double> TestHydro::ComputeForceHydrostatics() {
 	std::vector<double> position, displacement;
-	unsigned total_dofs = 6 * num_bodies;
-	position.resize(total_dofs, 0.0);
-	displacement.resize(total_dofs, 0.0);
+	assert(num_bodies > 0);
 
-	// orientation initialized to system current pos/rot vectors
 	for (int b = 0; b < num_bodies; b++) {
-		for (int i = 0; i < 3; i++) {
-			unsigned b_offset = 6 * b;
-			if (b_offset + i + 3 > total_dofs || b_offset + i < 0) {
-				std::cout << "temp index in hydrostatic force is bad " << std::endl;
-			}
-			position[i + b_offset] = bodies[b]->GetPos()[i];
-			position[i + 3 + b_offset] = bodies[b]->GetRot().Q_to_Euler123()[i]; // TODO check if we want this, or GetWvel_par()?
+		// initialize variables
+		auto body = bodies[b];
+		auto& body_h5file = file_info[b];
```

Comment:
Since this is an open source library, I would like to prioritize readability, and I would like to avoid using auto unless the type is explicitly clear. (Examples of this are in the demos when auto is used on a line when a constructor is called like `auto prismatic = chrono_types::make_shared<ChLinkLockPrismatic>();`)
I think if you change typenames, it is important to change them everywhere (VisualStudio has the rename tool to assist this change), and ensure the change is appropriate in every situation. If you change something with auto, there is a chance your code returns something unexpected and possibly just wrong.
As for your point about performance, I see what you are saying here but I still value readability over the tiny performance improvements to be gained here. I think if someone new were to start using this library, these lines would not be very clear.
Finally, most editors have ways to detect typos and notice if you are using a type that doesn't exist, some also have predictive text options to help with efficiency without compromising readability.

Reply:
I understand your perspective, it works for me, I explicitly declared the variables in the `ComputeHydrostatics` function in the last commit.